### PR TITLE
Add option for Cluster DNS having a FQDN other than cluster.local

### PR DESCRIPTION
**Pull request by barakmich**
Step one for poseidon/typhoon#19

### CODE REVIEW

**Comment by dghubble** [comment]
Not plausible until etcd-operator https://github.com/coreos/etcd-operator/releases/tag/v0.5.1 which has some breaking changes and I'd like that to be accepted in bootkube as well.

**Comment by barakmich** [comment]
If I wanted to try it myself, could I bump the etcd-operator container
version and add the flag?

**Comment by barakmich** [comment]
The generated certs would have the right SAN anyway :)

**Comment by dghubble** [comment]
Sure, go for it.

**Comment by barakmich** [comment]
Also, the operator is only on experimental-self-hosted-etcd

**Comment by dghubble** [comment]
Experimental self-hosted etcd is used on cloud platforms at the moment, although I'm moving to deprecate it https://github.com/poseidon/typhoon/issues/13.

In the mean time, I think on the next bootkube release, self-hosted etcd will get bumped, and we can start supporting this across platforms. We can bump it ourselves too, but I expect there will be a release fairly soon so we don't need to diverge.

**Comment by dghubble** [comment]
Self-hosted etcd is no longer used in Typhoon and self-hosted etcd assets have been deleted in #30.

File: tls-etcd.tf [hunk 1]
```diff
@@ -104,8 +104,8 @@ resource "tls_cert_request" "client" {
     var.etcd_servers,
     list(
       "localhost",
-      "*.kube-etcd.kube-system.svc.cluster.local",
-      "kube-etcd-client.kube-system.svc.cluster.local",
+      "*.kube-etcd.kube-system.svc.${var.cluster_dns_fqdn}",
```
Comment (dghubble) [inline tls-etcd.tf]: With self-hosted etcd gone, we should be able to delete these extra DNS SANs in this file entirely.

**Comment by dghubble** [comment]
Removed those SANs in https://github.com/poseidon/terraform-render-bootkube/pull/31/commits/5763b447dedd3330a62f076ff69b2fa834d9b088. Verified a multi-controller (i.e. multi-etcd) cluster's etcd setup was ok:

```
sudo etcdctl --ca-file=/etc/ssl/etcd/etcd-client-ca.crt --cert-file=/etc/ssl/etcd/etcd-client.crt --key-file=/etc/ssl/etcd/etcd-client.key --endpoint=https://127.0.0.1:2379 cluster-health
```

```
kubectl get cs
```

Can you rebase?

**Comment by barakmich** [comment]
Rebase was as easy as accepting your new version 👍

**Comment by dghubble** [comment]
http://blog.kubernetes.io/2017/04/configuring-private-dns-zones-upstream-nameservers-kubernetes.html makes me think it should be called `cluster_dns_suffix`, maybe as a followup.

**Pull request merged**